Fix Zammad plugin by filtering tickets after search

come-nc opened the pull request:
The API query is not searching correctly and is returning more tickets that it should, so we loop on them and get articles from the API to filter which one were actually updated in the time range.

This reduces false positive a lot.

psss commented on did/plugins/zammad.py:
```diff
@@ -61,7 +63,25 @@ def search(self, query):
+        except urllib.error.URLError as error:
+            log.debug(error)
+            raise ReportError(
+                "Zammad search on {0} failed.".format(self.url))
```
This part seems to be copy & pasted from the `search()` method. What about separating the common code to prevent duplication? There could be a common `search()` method used by both `get_articles()` and `get_tickets()`. What do you think, @come-nc?

psss reviewed (dismissed):
Thanks for improving this! Looks good, just two minor suggestions. Consider enabling `pre-commit` to catch code style errors early, see also the [develop](https://fmf.readthedocs.io/en/latest/contribute.html#develop) section.

sandrobonazzola commented:
@come-nc can you please rebase and address @psss comments?

come-nc commented:
> @come-nc can you please rebase and address @psss comments?

Rebased and fixed what I could, but it still complains the line is too long. I can rename the var `updated_at` into `u` but that’s kind of stupid?

sandrobonazzola reviewed (approved):
The plugin itself would benefit of some additional exception handling but can be done in separate patch.
